src/Marker.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  Marker.swift
+//  ios-sdk
+//
+//  Created by Sarah Lensing on 3/22/17.
+//  Copyright © 2017 Mapzen. All rights reserved.
+//
+
+import Foundation
+import TangramMap
+
+public enum MarkerType : Int {
+  case currentLocation, searchPin, routeLine
+}
+
+@objc(MZMarker)
+public class Marker: NSObject {
```

Comment:
This is a good start. I would generalize this to define a Marker Protocol that this class would then implement. You could even move much of that implementation into a protocol extension if you wanted.

Then, change out the implementing call sites for this class to use the protocol object instead of the actual class itself.
Thoughts?